File: WikiCodeParser/Tags/CodeTag.cs
﻿using System.Collections.Generic;
using WikiCodeParser.Nodes;

namespace WikiCodeParser.Tags
{
    public class CodeTag : Tag
    {
        public override string Token => "code";
        public override string Element => "code";

        public override INode FormatResult(Parser parser, State state, string scope, Dictionary<string, string> options, string text)
        {
            return new HtmlNode("<code>", new PlainTextNode(text), "</code>");
        }
    }
}

File: src/Tags/CodeTag.ts
import { Parser } from '..';
import { HtmlNode } from '../Nodes/HtmlNode';
import { INode } from '../Nodes/INode';
import { UnprocessablePlainTextNode } from '../Nodes/UnprocessablePlainTextNode';
import { ParseData } from '../ParseData';
import { State } from '../State';
import { Tag } from './Tag';

export class CodeTag extends Tag {
    constructor() {
        super();
        this.Token = 'code';
        this.Element = 'code';
    }

    public override FormatResult(_parser: Parser, _data: ParseData, _state: State, _scope: string, _options: Record<string, string>, text: string): INode {
        return new HtmlNode('<code>', new UnprocessablePlainTextNode(text), '</code>');
    }
}
